packages/suite/src/actions/firmware/firmwareActions.ts

```diff
@@ -190,30 +215,29 @@ export const validateFirmwareHash =
                 },
                 challenge: firmwareChallenge,
             });
+
+            // we don't want to have false negatives but more importantly false positives.
+            // Cases that should not lead to the big red error:
+            // - device disconnected, broken cable
+            // - errors from TrezorConnect in general
+            // Cases that should lead to the big red error:
+            // - device was unable to process 'GetFirmwareHash' message ('Unknown message')
+            // - errors from device in general
             if (!fwHash.success) {
-                dispatch({
-                    type: FIRMWARE.SET_ERROR,
-                    payload: `${fwHash.payload.error}. Unable to validate firmware hash. If you want to check authenticity of newly installed firmware please proceed to device settings and reinstall firmware.`,
-                });
-                analytics.report({
-                    type: EventType.FirmwareValidateHashError,
-                    payload: {
-                        error: fwHash.payload.error,
-                    },
-                });
+                // Device error
+                // todo: add a more generic way how to handle all device errors
+                if (fwHash.payload.error === 'Unknown message') {
```

Comment:
device errors should have payload.code

Reply:
thanks, will check that. before using that we should be pretty sure that code can't appear in any other case.

Reply:
unfortunately it will, TrezorConnect [uses own codes](https://github.com/trezor/trezor-suite/blob/develop/packages/connect/src/constants/errors.ts#L1) but we also know all possible firmware codes from[ protobuf messages](https://github.com/trezor/trezor-suite/blob/develop/packages/transport/messages.json#L2373-L2387)

Reply:
discussed in person, resolving